projects/core/directives/number-format/number-format.directive.ts

```diff
@@ -1,26 +1,28 @@
-import {Directive, forwardRef, Inject, Input} from '@angular/core';
+import {Directive, forwardRef, Inject, Input, SkipSelf} from '@angular/core';
 import {TuiNumberFormatSettings} from '@taiga-ui/core/interfaces';
-import {TUI_NUMBER_FORMAT, TUI_NUMBER_FORMAT_OBSERVABLE} from '@taiga-ui/core/tokens';
+import {TUI_NUMBER_FORMAT} from '@taiga-ui/core/tokens';
 import {BehaviorSubject} from 'rxjs';
 
 @Directive({
     selector: '[tuiNumberFormat]',
     providers: [
         {
-            provide: TUI_NUMBER_FORMAT_OBSERVABLE,
+            provide: TUI_NUMBER_FORMAT,
             useExisting: forwardRef(() => TuiNumberFormatDirective),
         },
     ],
 })
 export class TuiNumberFormatDirective extends BehaviorSubject<TuiNumberFormatSettings> {
     @Input()
     set tuiNumberFormat(format: Partial<TuiNumberFormatSettings>) {
-        this.next({...this.settings, decimalLimit: NaN, ...format});
+        this.next({...this.settings, ...format});
     }
 
     constructor(
-        @Inject(TUI_NUMBER_FORMAT) private readonly settings: TuiNumberFormatSettings,
+        @SkipSelf()
+        @Inject(TUI_NUMBER_FORMAT)
+        private readonly settings: TuiNumberFormatSettings,
```

Comment:
That will be tricky once `TUI_NUMBER_FORMAT` is always and `Observable` but we can do it. Basically, it will no longer be a `BehaviorSubject` and instead there will be `combineLatest` of upper number format and an internal `ReplaySubject(1)` that we will `next` to in setter.

Reply:
rewritten :+1: I hope I got you right